spec/requests/api/v2/exchange_rates/files_controller_spec.rb

```diff
@@ -54,5 +55,39 @@
         expect(response.body).to eq(data)
       end
     end
+
+    context 'when requesting HMRC CSV format' do
+      let(:type) { 'monthly_csv_hmrc' }
+      let(:format) { 'csv' }
+
+      before { get api_exchange_rates_file_path("#{type}_#{year}-#{month}", format: :csv) }
+
+      it 'returns HTTP status :ok' do
+        expect(response).to have_http_status(:ok)
+      end
+
+      it 'returns content type text/csv; charset=utf-8' do
+        expect(response.content_type).to eq('text/csv; charset=utf-8')
+      end
+
+      it 'returns the correct Content-Disposition header' do
+        expect(response.headers['Content-Disposition']).to eq("attachment; filename=monthly_csv_hmrc_#{year}-#{month}.csv")
+      end
+
+      it 'returns the CSV data as the response body' do
+        expect(response.body).to eq(data)
+      end
+    end
+
+    context 'when requesting invalid type' do
+      let(:type) { 'non_existing_type' }
+      let(:format) { 'csv' }
```

Comment:
dont think this is used?

Reply:
that's a test case to test a failing condition